database.py

```diff
@@ -88,15 +92,52 @@ def get_trusted(self, userid: int) -> bool:
 			c.execute('''SELECT trusted FROM users WHERE userid = ?''', (userid,))
 			return bool(c.fetchone()[0])
 
-	def cleanup_votekicks(self):
+	def add_vk_messages(self, bad_user: int, msgs: list[int]):
+		'''
+		Adds `msgs` to the list of messages associated
+		with `bad_user`'s votekick.
+		'''
+		with self.mutex:
+			for msg in msgs:
+				self.db.execute(
+					'''INSERT INTO vk_messages VALUES (?, ?)''',
+					(bad_user, msg)
+				)
+			self.db.commit()
+
+	def end_votekick(self, bad_user: int):
+		'''
+		Removes data about `bad_user`'s votekick.
+		Called after a successful votekick.
+		'''
+		with self.mutex:
+			c = self.db.cursor()
+			c.execute('''DELETE FROM votekicks where bad_user=?''', (bad_user,))
+			self.db.commit()
+
+	def cleanup_votekicks(self) -> list[int]:
+		'''
+		Cleans up outdated votekick data from the database
+		and returns the list of messages to be removed.
+		'''
 		with self.mutex:
 			c = self.db.cursor()
 			c.execute('''DELETE FROM votekicks WHERE timeout < JULIANDAY('NOW')''')
-			c.fetchall()
+			c.execute('''
+				SELECT msg FROM vk_messages
+				WHERE bad_user NOT IN (
+					SELECT bad_user FROM votekicks
+				);''')
+			msgs = [row[0] for row in c.fetchall()]
+			c.execute('''

```

Comment:
we should probably use the `msg` variable instead of selecting again, just to make sure the data stays congruent